lib/msal-browser/src/interaction_handler/InteractionHandler.ts

```diff
@@ -190,62 +156,29 @@ export class InteractionHandler {
             PerformanceEvents.AuthClientAcquireToken,
             this.logger,
             this.performanceClient,
-            this.authCodeRequest.correlationId
+            request.correlationId
         )(this.authCodeRequest, authCodeResponse)) as AuthenticationResult;
-        this.browserStorage.cleanRequestByState(state);
         return tokenResponse;
     }
 
     /**
-     * Updates authority based on cloudInstanceHostname
```

Comment:
Curious: Why were we making an extra network call for instance, are we assuming authority changes? Hopefully the multi-tenant feature should cover this.

Reply:
Ah! never mind. I see we need this when we make the request which you moved below. I am interested to check if we can avoid this when applicable, as tenant-id switch should not change anything with cloud discovery.

Reply:
This is different from tenant switch. This is switching the _domain_ of the authority, I believe for outage scenarios